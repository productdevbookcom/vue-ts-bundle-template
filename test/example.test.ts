import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { Button } from '../src'

const slotText = 'Test content'

describe('button', () => {
  it('alert render', () => {
    const wrapper = mount(Button, {
      slots: {
        default: slotText,
      },
    })

    expect(wrapper.text()).toBe(slotText)
  })

  it('loading', () => {
    const wrapper = mount(Button, {
      props: {
        loading: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('disabled', () => {
    const wrapper = mount(Button, {
      props: {
        disabled: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
