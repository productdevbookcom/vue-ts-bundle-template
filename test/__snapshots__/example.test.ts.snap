// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > disabled 1`] = `"<button><span></span></button>"`;

exports[`button > disabled test 1`] = `"<button><span></span></button>"`;

exports[`button > loading 1`] = `"<button><span class=\\"h-12 w-20 h-full\\"> hello </span></button>"`;

exports[`button > loading test 1`] = `"<button><span class=\\"h-12 w-20 h-full\\"> hello </span></button>"`;
